src/load_mid.cpp

```diff
@@ -1149,8 +1149,8 @@ static void mid_prog_to_notes(MIDTRACK *tp, ULONG adjust, ULONG tmin)
 		}
 	}
 	if( ton > toff ) {
-		char info[40];
-		sprintf(info,"channel %d, %ld > %ld note %d", tp->chan + 1, (long)ton, (long)toff, n);
+		char info[128];
+		snprintf(info,sizeof(info),"channel %d, %ld > %ld note %d", tp->chan + 1, (long)ton, (long)toff, n);
```

Comment:
I applied this locally to support older MSVC and other toolchains, to be safe.
(Also attaching as a patch file: [patch.txt](https://github.com/Konstanty/libmodplug/files/6680681/patch.txt))
```diff
diff --git a/src/load_mid.cpp b/src/load_mid.cpp
index 4a93322..caf5b2c 100644
--- a/src/load_mid.cpp
+++ b/src/load_mid.cpp
@@ -1125,6 +1125,12 @@ static void mid_notes_to_percussion(MIDTRACK *tp, ULONG adjust, ULONG tmin)
 	if( lno && lno->next ) mid_stripoff(tp, lno);
 }
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(__WACTOMC__) || \
+    (defined(__MINGW32__) && !(defined(__USE_MINGW_ANSI_STDIO) && __USE_MINGW_ANSI_STDIO != 0))
+#define BAD_SNPRINTF
+#undef snprintf
+#define snprintf _snprintf
+#endif
 static void mid_prog_to_notes(MIDTRACK *tp, ULONG adjust, ULONG tmin)
 {
 	MIDEVENT *e, *lno = 0;
@@ -1174,6 +1180,9 @@ static void mid_prog_to_notes(MIDTRACK *tp, ULONG adjust, ULONG tmin)
 	if( ton > toff ) {
 		char info[128];
 		snprintf(info,sizeof(info),"channel %d, %ld > %ld note %d", tp->chan + 1, (long)ton, (long)toff, n);
+		#ifdef BAD_SNPRINTF
+		info[sizeof(info)-1] = '\0';
+		#endif
 		mid_message("melody track ends with note on (%s)", info);
 	}
 	if( lno && lno->next ) mid_stripoff(tp, lno);
```

Reply:
I think I'd be OK with just unconditionally adding that nul (I usually do, but forgot it here).

Reply:
Actually, since the extra define would still need to be done with `snprintf`, this could probably just be an `sprintf` (the buffer should be more than large enough now).

Reply:
Even better, thanks.